### ISSUE TITLE
Modular Router Dupe

### ISSUE THREAD

**Issue by syrun97** [DarkPacks/SevTech-Ages#2649]
## Bug Report

https://www.youtube.com/watch?v=bjFY7eGHyH4&feature=youtu.be
It Dupe Items when flinger/dropper module and vacuum module is installed in one router.
## Expected Behaviour

Not dupe items.
## Possible Solution

IDK
## Steps to Reproduce (for bugs)

1.Place Flinger/dropper module in router.
2.place Vacuum module in router.
3. Place any Item in buffer.
4. ???
5.PROFIT!!!

## Logs

* Client/Server Log:
[latest.log](https://github.com/DarkPacks/SevTech-Ages/files/2042294/latest.log)

* Crash Log:no crash

## Client Information

* Modpack Version:3.0.7 w/ integrated tunnels
* Java Version:1.8.0_171
* Launcher Used:twitch
* Memory Allocated:5G
* Server/LAN/Single Player:single player
* Optifine Installed:nope
* Shaders Enabled:nope

**Comment by syrun97** [comment]
Submited on Modular router github
https://github.com/desht/ModularRouters/issues/39

**Comment by desht** [comment]
This appears to be a mod incompatibility; it happens in Sevtech but not in my test instance or any other pack I've use Modular Routers in, or added Modular Routers to.

See https://github.com/desht/ModularRouters/issues/39 for more info.  I should be able to work around it in Modular Routers by being less trusting of the ``World#spawnEntity()`` return value.

**Comment by desht** [comment]
Pretty sure it's Realistic Item Drops triggering this; it cancels the EntityJoinWorldEvent that is fired when item entities spawn (and spawns its own items). Modular Routers interprets this event cancellation as a failure to spawn the item, so doesn't extract the item from the router buffer.

I can work around it in Modular Routers; there will be a new release very soon.

**Comment by desht** [comment]
So, it is Realistic Item Drops, but mainly because in Sevtech, Realistic Item Drops "Dupe Workaround" config has been switched off (the mod sets it to *true* by default):
```
   # A workaround for machines that check their drops before subtracting items. Disable if there are noticable side effects [default: true]
    B:"Dupe Workaround"=false
```

Modular Routers is one of those machines that checks its drops, hence the dupe.  I presume there's a good reason for Sevtech disabling it, but I don't know what that is.

In any case, I'm OK with making Modular Routers not check the result of ``World#spawnEntity()`` which gets around this problem.

**Comment by desht** [comment]
Fixed in Modular Routers 3.1.5 release (just approved) - https://minecraft.curseforge.com/projects/modular-routers/files/2567848
